File: 프로그래머스/lv1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        HashMap<String, Integer> map = new HashMap<>();
        for (int i = 0; i < participant.length; i++) {
            map.put(participant[i], map.getOrDefault(participant[i], 0) + 1);
        }
        for (int i = 0; i < completion.length; i++) {
            map.put(completion[i], map.get(completion[i]) - 1);
        }
        
        Set<String> keySet = map.keySet();
        
        for (String key : keySet) {
            if (map.get(key) == 1) answer = key;
        }
        return answer;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    list = {}
    for part in participant:
        list[part] = list.get(part, 0) + 1
    
    for comp in completion:
        list[comp] = list.get(comp, 0) - 1
    
    for k, v in list.items():
        if v > 0:
            return k